src/server/system_services/system_store.js

```diff
@@ -540,17 +540,18 @@ class SystemStore extends EventEmitter {
         let non_deleted_query = {
             deleted: null
         };
-        return mongo_client.instance().connect()
+        return db_client.instance().connect()
             .then(() => P.map(COLLECTIONS,
-                col => mongo_client.instance().collection(col.name)
-                .find(non_deleted_query)
-                .toArray()
-                .then(res => {
-                    for (const item of res) {
-                        this._check_schema(col, item, 'warn');
-                    }
-                    target[col.name] = res;
-                })
+                col => {
+                    let f = db_client.instance().collection(col.name).find(non_deleted_query);
+                    if (!config.USE_POSTGRESQL) f = f.toArray();
```

Comment:
@jeniawhite Can't you hide this inside the db_client? I wouldn't want the system_store to be filled with references to config.USE_POSTGRESQL. In general we will keep supporting both db modes, and possibly add another, so this should be clean from the caller side.